eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/ManagementOutputLibrary.cs

```diff
@@ -65,41 +65,57 @@ private static void InitializeResourceClients(
             ref IReadOnlyList<MockableResourceProvider>? _mockableClients,
             ref ExtensionProvider? _extensionProvider)
         {
-            if (_resourceClients is not null || _collectionClients is not null || _mockableClients is not null)
+            if (_resourceClients is not null ||
+                _collectionClients is not null ||
+                _mockableClients is not null ||
+                _extensionProvider is not null)
             {
                 return; // already initialized
             }
 
-            var metadatas = ManagementClientGenerator.Instance.InputLibrary.ResourceMetadatas;
-            var resources = new List<ResourceClientProvider>(metadatas.Count);
-            var collections = new List<ResourceCollectionClientProvider>(metadatas.Count);
-            foreach (var resourceMetadata in metadatas)
+            var resourceMetadatas = ManagementClientGenerator.Instance.InputLibrary.ResourceMetadatas;
+            var resourceMethodCategories = new Dictionary<ResourceMetadata, ResourceMethodCategory>(resourceMetadatas.Count);
+            // categorize the resource methods
+            foreach (var resourceMetadata in resourceMetadatas)
             {
-                var resource = ResourceClientProvider.Create(resourceMetadata);
-                resources.Add(resource);
+                var categorizedMehtods = resourceMetadata.CategorizeMethods();
```

Comment:
can we construct these in the same loop in L90?

Reply:
done